### ISSUE TITLE
Layer type: "Grid" in {deploy.prototxt}

### ISSUE THREAD

**Issue by JoseSoares96** [twhui/LiteFlowNet#59]
When I'm trying to use the network i get the following error:
"WARNING: Logging before InitGoogleLogging() is written to STDERR
F0114 15:31:50.987926 12987 layer_factory.hpp:81] Check failed: registry.count(type) == 1 (0 vs. 1) **Unknown layer type: Grid** 

> (known types: AbsVal, Accum, Accuracy, ArgMax, BNLL, BatchNorm, BatchReindex, Bias, BlackAugmentation, ChannelNorm, Concat, ContrastiveLoss, Convolution, Correlation, Correlation1D, Crop, CustomData, Data, DataAugmentation, Deconvolution, DisparityData, Downsample, Dropout, DummyData, ELU, Eltwise, Embed, EuclideanLoss, Exp, FLOWriter, Filter, Flatten, FloatReader, FloatWriter, FlowAugmentation, FlowWarp, GenerateAugmentationParameters, HDF5Data, HDF5Output, HingeLoss, Im2col, ImageData, ImgReader, ImgWriter, InfogainLoss, InnerProduct, Input, L1Loss, LRN, LSTM, LSTMUnit, Log, LpqLoss, MVN, Mean, MemoryData, MultinomialLogisticLoss, NegReLU, PFMWriter, PReLU, Parameter, Pooling, Power, RNN, ReLU, Reduction, Resample, Reshape, SPP, Scale, Sigmoid, SigmoidCrossEntropyLoss, Silence, Slice, Softmax, SoftmaxWithLoss, Split, TanH, Threshold, Tile, WindowData) 

for what i understood this layer type is not recognized.
**But i add the files that are mentioned in the readme file.**

> Feature warping (f-warp) layer:
> 
> The source files include /src/caffe/layers/warp_layer.cpp, /src/caffe/layers/warp_layer.cu, and /include/caffe/layers/warp_layer.hpp.
> 
> The grid pattern that is used by f-warp layer is generated by a grid layer. The source files include /src/caffe/layers/grid_layer.cpp and /include/caffe/layers/grid_layer.hpp.

**Probably i'm not doing some step.**

I'm working with : 
NVIDIA-SMI 440.44      
Driver Version: 440.44       
CUDA Version: 10.2 
Cudnn is the version 5.1
caffe version 1.0.0

Thanks a lot!

**Comment by JoseSoares96** [comment]
Hi,
I think i solved the first errors that i announce. Meanwhile i got another one:

> src/caffe/layers/expmax_layer.cpp:12:41: error: ‘class caffe::LayerParameter’ has no member named ‘expmax_param’; did you mean ‘argmax_param’?
>    const Dtype base = this->layer_param_.expmax_param().base();
>                       ~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~
>                       argmax_param

of course that this results in a error.

> Makefile:576: recipe for target '.build_release/src/caffe/layers/expmax_layer.o' failed

> make: *** [.build_release/src/caffe/layers/expmax_layer.o] Error 1

Do u know something about this and how to solve it?  Thanks again

**Comment by twhui** [comment]
I can compile the Caffe package of LiteFlowNet without any errors. There may be some problems in your code.

**Comment by JoseSoares96** [comment]
I can compile it too. I just wanted to compile the layers that i need to another caffe project. But i found a way around. Thanks.
Closing the topic.
